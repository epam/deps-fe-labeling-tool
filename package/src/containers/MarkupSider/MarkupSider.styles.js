import styled from 'styled-components'
import { Sider } from '@/components/Sider'
import { COLORS } from '@/theme/theme.default'

const MarkupSidebarSider = styled(Sider)`
  border-left: 1px solid ${COLORS.SECONDARY};
  z-index: 1;
`
export { MarkupSidebarSider }
