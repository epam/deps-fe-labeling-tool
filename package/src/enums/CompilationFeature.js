const CompilationFeature = {
  SHOW_NOT_IMPLEMENTED: 'SHOW_NOT_IMPLEMENTED'
}

export {
  CompilationFeature
}
