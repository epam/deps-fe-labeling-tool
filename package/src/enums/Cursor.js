const Cursor = {
  DEFAULT: 'default',
  GRABBING: 'grabbing',
  HOVER: 'hover',
  MOVE: 'move'
}

export {
  Cursor
}
