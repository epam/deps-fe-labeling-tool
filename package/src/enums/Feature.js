const Feature = {
  AUTO_SAVE: 'autoSave',
  EXPORT: 'export',
  IMPORT: 'import',
  PAGING: 'paging',
  LANGUAGE: 'language',
  ROTATION: 'rotation',
  SPECIAL_SYMBOLS: 'specialSymbols',
  MANAGE_FIELDS: 'manageFields',
  MULTI_ASSIGN_LABELS: 'multiAssignLabels',
  MULTI_ASSIGN_TABLES: 'multiAssignTables'
}

export {
  Feature
}
