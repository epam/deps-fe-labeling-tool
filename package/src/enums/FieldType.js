const FieldType = {
  CHECKMARK: 'checkmark',
  ENUM: 'enum',
  LIST: 'list',
  PAIR: 'pair',
  STRING: 'string',
  TABLE: 'table',
  DATE: 'date'
}

const RESOURCE_FIELD_TYPE = {
  [FieldType.STRING]: 'String',
  [FieldType.ENUM]: 'Enumeration',
  [FieldType.CHECKMARK]: 'Checkbox',
  [FieldType.LIST]: 'List',
  [FieldType.PAIR]: 'Key Value Pair',
  [FieldType.TABLE]: 'Table',
  [FieldType.DATE]: 'Date'
}

export {
  FieldType,
  RESOURCE_FIELD_TYPE
}
