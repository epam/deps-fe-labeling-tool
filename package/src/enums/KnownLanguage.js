const KnownLanguage = {
  ENGLISH: 'eng',
  RUSSIAN: 'rus',
  CHINESE_SIMPLIFIED: 'chi_sim',
  DEUTSCH: 'deu',
  SPANISH: 'spa',
  UKRAINIAN: 'ukr'
}

export {
  KnownLanguage
}
