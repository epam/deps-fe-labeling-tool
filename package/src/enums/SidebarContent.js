const SidebarContent = {
  MARKUP: 'MARKUP',
  TABLE_DATA: 'TABLE_DATA'
}

const RESOURCE_SIDEBAR_TAB = {
  [SidebarContent.MARKUP]: 'Properties',
  [SidebarContent.TABLE_DATA]: 'Table Data'
}

export {
  SidebarContent,
  RESOURCE_SIDEBAR_TAB
}
