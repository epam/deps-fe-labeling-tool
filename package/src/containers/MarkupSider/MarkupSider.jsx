import React, { useEffect, useMemo } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { MarkupSidebar } from '@/components/MarkupSidebar'
import { HotKeyEvent } from '@/constants/hotKeys'
import { TableDataSidebar } from '@/containers/TableDataSidebar'
import { SidebarContent } from '@/enums/SidebarContent'
import { withHotKeys } from '@/hocs/withHotKeys'
import { activeSidebarSelector } from '@/selectors/ui'
import { MarkupSidebarSider } from './MarkupSider.styles'

const WIDTH_RIGHT_SIDER = 300
const ZERO_WIDTH = 0

const MarkupSider = ({ registerHandlers, setMarkupSiderWidth, width, activeSidebar }) => {
  const hotKeyHandlers = useMemo(() => ({
    [HotKeyEvent.COLLAPSE_EXPAND_SIDERS]: () => {
      if (width) {
        setMarkupSiderWidth(ZERO_WIDTH)
      } else {
        setMarkupSiderWidth(WIDTH_RIGHT_SIDER)
      }
    }

  }), [width, setMarkupSiderWidth])

  const Sidebar = useMemo(() => {
    switch (activeSidebar) {
      case SidebarContent.TABLE_DATA:
        return <TableDataSidebar />
      case SidebarContent.MARKUP:
        return <MarkupSidebar />
    }
  }, [activeSidebar])

  useEffect(() => {
    registerHandlers(hotKeyHandlers)
  }, [hotKeyHandlers, registerHandlers])

  if (!width) {
    return null
  }

  return (
    <MarkupSidebarSider
      width={WIDTH_RIGHT_SIDER}
    >
      {Sidebar}
    </MarkupSidebarSider>
  )
}

MarkupSider.propTypes = {
  registerHandlers: PropTypes.func.isRequired,
  activeSidebar: PropTypes.oneOf(Object.values(SidebarContent)).isRequired,
  setMarkupSiderWidth: PropTypes.func.isRequired,
  width: PropTypes.number.isRequired
}

const mapStateToProps = (state) => ({
  activeSidebar: activeSidebarSelector(state)
})

const MarkupSiderWithHotKeys = withHotKeys(connect(mapStateToProps)(MarkupSider))

export {
  MarkupSiderWithHotKeys as MarkupSider,
  WIDTH_RIGHT_SIDER
}
