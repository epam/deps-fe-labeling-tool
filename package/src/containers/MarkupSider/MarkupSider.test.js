import React from 'react'
import { shallow } from 'enzyme'
import { MarkupSider } from '@/containers/MarkupSider'
import { SidebarContent } from '@/enums/SidebarContent'

jest.mock('@/hocs/withHotKeys', () => ({
  withHotKeys: (Component) => Component
}))

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn((f) => f())
}))

const { WrappedComponent } = MarkupSider

describe('Container: MarkupSider', () => {
  let wrapper, defaultProps

  beforeEach(() => {
    defaultProps = {
      setMarkupSiderWidth: jest.fn(),
      activeSidebar: SidebarContent.MARKUP,
      width: 1500,
      registerHandlers: jest.fn()
    }

    jest.clearAllMocks()
    wrapper = shallow(<WrappedComponent {...defaultProps} />)
  })

  it('should render layout correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return null when prop.width is 0', () => {
    wrapper.setProps({ ...defaultProps, width: 0 })
    expect(wrapper.props()).toEqual({})
  })

  it('should call props registerHandlers when component is mounted', () => {
    expect(defaultProps.registerHandlers).nthCalledWith(1, { COLLAPSE_EXPAND_SIDERS: expect.any(Function) })
  })
})
