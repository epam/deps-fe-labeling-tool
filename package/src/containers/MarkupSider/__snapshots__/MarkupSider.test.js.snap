// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: MarkupSider should render layout correctly 1`] = `
<MarkupSiderstyles__MarkupSidebarSider
  width={300}
>
  <MarkupSidebar />
</MarkupSiderstyles__MarkupSidebarSider>
`;
